### PR TITLE
HCAP-708

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
* Added error page for when a user attempts to access a participant record that they should not have access to
  * Error page has a button that leads back to the home page
* Confirm interest button is hidden after submission
* Added confirm interest flow to ParticipantEOI detail page

### CODE REVIEW

File: server/services/participants.js [hunk 1]
```diff
@@ -612,7 +612,46 @@ const getParticipantByIdWithStatus = async ({ id, userId }) =>
     })
     .find({ id, 'user.user_id': userId });
 
+const setParticipantLastUpdated = async (id) => {
+  // Find participants
+  let participant = (await getParticipantByID({ id }))[0];
+  // Don't change status if participant is withdrawn
+  if (participant.interested !== 'withdrawn') {
+    // Only change history if the interested column isn't yes
+    if (participant.intereted !== 'yes') {
```
Comment (Zenen-freshworks) [inline server/services/participants.js:621]:
```suggestion
    if (participant.interested !== 'yes') {
```

File: server/server.js [hunk 2]
```diff
@@ -281,12 +281,7 @@ app.get(
 
 app.get(
   `${apiBaseUrl}/participant`,
-  keycloak.allowRolesMiddleware(
-    'health_authority',
-    'ministry_of_health',
-    'employer',
-    'participant'
```
[the remark this answers is not in the document]
Comment (dbayly-freshworks) [reply server/server.js:288]: We do not. There's a similar function that gets user information in `participant-user.js` that has a step to verify that a participant has permission to view that record.

File: server/services/participants.js [hunk 3]
```diff
@@ -612,7 +612,46 @@ const getParticipantByIdWithStatus = async ({ id, userId }) =>
     })
     .find({ id, 'user.user_id': userId });
 
+const setParticipantLastUpdated = async (id) => {
+  // Find participants
+  let participant = (await getParticipantByID({ id }))[0];
```
Comment (Zenen-freshworks) [inline server/services/participants.js:617]:
```suggestion
  let [participant] = await getParticipantByID({ id });
```
nitpick: I like this format more for getting an entry in an array

**Review by Zenen-freshworks — DISMISSED** [review]

File: server/services/participants.js (hunk 3 above)
Comment (dbayly-freshworks) [reply server/services/participants.js:617]: 10/10 Looks way better.

File: client/src/pages/private/ParticipantEOI.js [hunk 4]
```diff
@@ -145,8 +165,28 @@ export default () => {
...
+    if (success) {
+      openToast({
+        status: ToastStatus.Success,
+        message: 'Confirm interest success',
+      });
+      setDisableConfirmInterest(true);
+    } else {
+      openToast({
+        status: ToastStatus.Error,
+        message: 'Confirm interest failure',
```
Comment (fwkendall) [inline client/src/pages/private/ParticipantEOI.js:181]: I'm adding getting toast message copy finalized to the prod readiness ticket FYI - so we don't need to worry about this for now

File: client/src/pages/private/ParticipantEOI.js [hunk 5]
```diff
@@ -189,6 +250,21 @@ export default () => {
             </Button>
           </DialogActions>
         </Dialog>
+        <Dialog title='Confirm Interest' open={openConfirmInterest} onClose={onClose}>
+          <DialogContent>
+            <Typography variant='subtitle2'>
+              By clicking Confirm, you agree to proceed with this form to apply for HCAP.
```
Comment (fwkendall) [inline client/src/pages/private/ParticipantEOI.js:256]: also needs copy for prod (should not be this)

File: server/routes/participant-user.js [hunk 6]
```diff
@@ -123,4 +126,18 @@ router.post(
...
+  '/participant/:id/reconfirm_interest',
+  asyncMiddleware(async (req, res) => {
+    const { user_id: userId } = req.user;
+    const { id } = req.params;
+    const participants = await getParticipantByIdWithStatus({ id, userId });
+    if (!participants.length) {
+      return res.status(401).send({ message: 'You do not have permission to view this record' });
+    }
+    await setParticipantLastUpdated(id);
+    return res.status(201).send({ message: 'Reconfirm interest successful.' });
```
Comment (fwkendall) [inline server/routes/participant-user.js:139]: nice 201 👍

**Review by fwkendall — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

**Pull request merged**